core/src/App.html

```diff
@@ -304,22 +301,34 @@
           }
         });
         window.addEventListener('message', async e => {
+          const allIframes = [...Iframe.getAllIframes(node, this.get().modalIframe), {contentWindow: window, luigi: {viewUrl: window.location.href, trustedDomain: window.location.origin}}];
+          const isOriginWindow = (iframe) => iframe && (e.source === iframe.contentWindow);
+          const iframe = allIframes.find(isOriginWindow);
+          if (!(iframe && iframe.luigi && iframe.luigi.viewUrl)) return;
+
+          if ('luigi.navigate.ok' === e.data.msg) {
+            if (iframe.luigi.nextViewUrl && iFrameHelpers.urlMatchesTheDomain(iframe.luigi.nextViewUrl, e.origin)) {
+              iframe.luigi.viewUrl = iframe.luigi.nextViewUrl;
+              iframe.luigi.nextViewUrl = '';
+            }
+            config.navigateOk = true;
+          }
+
+          if (!iFrameHelpers.iframeIsSameDomain(iframe, e.origin)) return;
+
           if ('luigi.get-context' === e.data.msg) {
-            if (
-              this.get().modalIframe &&
-              e.source === this.get().modalIframe.contentWindow
-            ) {
+            if (isOriginWindow(this.get().modalIframe)) {
               let ctx = this.get().modalIframeData.context;
               const modalConfig = {
                 ...config,
                 iframe: this.get().modalIframe,
-                context: ctx,
+                context: ctx, 
                 pathParams: this.get().modalIframeData.pathParams,
                 nodeParams: this.get().modalIframeData.nodeParams,
                 modal: true
               };
               sendContextToClient(this, modalConfig, {});
-            } else if (config.iframe) {
+            } else if (isOriginWindow(iframe)) {
```

Comment:
As iframe really contains the source of the message, I would say here we just need `else`.  Above we are already extracting the iframe from all the iframes using `isOriginWindow` so we are sure `iframe` here is the right one and we don't need to check again.